vms/evm/warp/service.go

```diff
@@ -0,0 +1,234 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package warp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/libevm/common/hexutil"
+	"github.com/ava-labs/libevm/log"
+
+	"github.com/ava-labs/avalanchego/cache"
+	"github.com/ava-labs/avalanchego/cache/lru"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p/acp118"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs/executor"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp/payload"
+)
+
+var errNoValidators = errors.New("cannot aggregate signatures from subnet with no validators")
+
+// API introduces snowman specific functionality to the evm.
+// It provides caching and orchestration over the core warp primitives.
+type API struct {
+	chainContext        *snow.Context
```

Comment:
Can we remove this and just manually pass in the required fields? `snow.Context` knows about a lot of consensus state that we should try to minimize usage of. Even if we kept this - this should be `snowContext` - we should avoid trying to invent new language.